src/main/scala/dpla/ingestion3/mappers/providers/NaraExtractor.scala

```diff
@@ -0,0 +1,256 @@
+
+    if (parentRecordGroupIds.nonEmpty) parentRecordGroupIds else parentCollectionIds
+  }
+
+  def contributor(xml: NodeSeq): Seq[String] = {
+
+    //TODO: not handling multiple display values. haven't found example yet.
+
+    val organizationalContributors = for {
+      org <- xml \\ "organizationalContributorArray" \ "organizationalContributor"
+      name = (org \ "contributor" \ "termName").text
+      _type = (org \ "contributorType" \ "termName").text
+      if _type != "Publisher"
+    } yield name
+
+    val personalContributors = for {
+      person <- xml \\ "personalContributorArray" \ "personalContributor"
+      name = (person \ "contributor" \ "termName").text
+      //_type = (person \ "contributorType" \ "TermName").text
+    } yield name
+
+    organizationalContributors ++ personalContributors
+  }
+
+  def creator(xml: NodeSeq): Seq[String] = {
+
+    //TODO: not handling multiple display values. haven't found example yet.
+    val organizationalCreators = for (
+      name <- xml \\ "creatingOrganizationArray" \ "creatingOrganization" \ "creator" \ "termName"
+    ) yield name.text
+
+    val individualCreators = for (
+      name <- xml \\ "creatingIndividualArray" \ "creatingIndividual" \ "creator" \ "termName"
+    ) yield name.text
+
+    if (organizationalCreators.nonEmpty) organizationalCreators else individualCreators
+  }
+
+
+  def nodeToDateString(nodeOption: Option[Node]): Option[String] = nodeOption match {
+    case None => None
+    case Some(node) =>
+      val year = (node \ "year").text
+      val month = (node \ "month").text
+      val day = (node \ "day").text
+
+      (year, month, day) match {
+        case (y, m, d) if y.isEmpty => None
+        case (y, m, d) if m.isEmpty => Some(year)
+        case (y, m, d) if d.isEmpty => Some(s"$year-$month")
+        case (y, m, d) => Some(s"$year-$month-$day")
+      }
+  }
+
+  def getDisplayDate(start: Option[Node], end: Option[Node]): Option[String] = {
+    if (start.isEmpty && end.isEmpty) {
+      None
+    } else {
+      val startString = start.map(_.text).getOrElse("unknown")
+      val endString = end.map(_.text).getOrElse("unknown")
+      Some(s"$startString/$endString")
+    }
+  }
+
+  def simpleDate(nodeSeq: NodeSeq): Seq[EdmTimeSpan] =
+    nodeSeq.map(node => EdmTimeSpan(originalSourceDate = nodeToDateString(Some(node))))
+
+  def date(xml: NodeSeq): Seq[EdmTimeSpan] = {
+
+    val coverageDates = for {
+      coverageDate <- xml \\ "coverageDates"
+      coverageStartDate = (coverageDate \ "coverageStartDate").headOption
+      coverageEndDate = (coverageDate \ "coverageEndDate").headOption
+      //dateQualifier = coverageDate \ "dateQualifier" //todo not sure what to do with qualifier in EDTF
+      if coverageStartDate.nonEmpty || coverageEndDate.nonEmpty
+
+    } yield {
+      val displayDate = getDisplayDate(coverageStartDate, coverageEndDate)
+      EdmTimeSpan(
+        originalSourceDate = displayDate, //this is cheating?
```

Comment:
`sourceResource.date` is supposed to be for date of publication or date of creation, as I understand it, whereas `sourceResourc.temporal` is for "aboutness." Therefore, with regard to whether this is cheating, I'd just say it's not if you can confirm that these "coverage dates" are for the date the source resource was generated or published.